pkg/controller/storagecluster/reconcile.go

```diff
@@ -717,16 +779,37 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, re
 		statusutil.MapCephClusterNoConditions(&r.conditions, reason, message)
 	} else {
 		// Interpret CephCluster status and set any negative conditions
-		statusutil.MapCephClusterNegativeConditions(&r.conditions, found)
+		if sc.Spec.ExternalStorage.Enable {
+			statusutil.MapExternalCephClusterNegativeConditions(&r.conditions, found)
+		} else {
+			statusutil.MapCephClusterNegativeConditions(&r.conditions, found)
+		}
 	}
 
 	// When phase is expanding, wait for CephCluster state to be updating
 	// this means expansion is in progress and overall system is progressing
 	// else expansion is not yet triggered
 	if sc.Status.Phase == statusutil.PhaseClusterExpanding &&
+		!sc.Spec.ExternalStorage.Enable &&
```

Comment:
```suggestion
```
If `sc.Status.Phase == statusutil.PhaseClusterExpanding` is `true`,  `!sc.Spec.ExternalStorage.Enable` is also `true` because we never do expansion for External Cluster.

Reply:
Good point.

Reply:
Yup, removed the line...